homeassistant/components/onvif/button.py

```diff
@@ -39,3 +41,34 @@ async def async_press(self) -> None:
         """Send out a SystemReboot command."""
         device_mgmt = self.device.device.create_devicemgmt_service()
         await device_mgmt.SystemReboot()
+
+
+class SetSystemDateAndTimeButton(ONVIFBaseEntity, ButtonEntity):
+    """Defines a ONVIF SetSystemDateAndTime button."""
+
+    _attr_entity_category = ENTITY_CATEGORY_CONFIG
+
+    def __init__(self, device: ONVIFDevice) -> None:
+        """Initialize the button entity."""
+        super().__init__(device)
+        self._attr_name = f"{self.device.name} Set System Date and Time"
+        self._attr_unique_id = f"{self.device.info.mac or self.device.info.serial_number}_setsystemdatetime"
+
+    async def async_press(self) -> None:
+        """Send out a SetSystemDateAndTime command."""
+        device_mgmt = self.device.device.create_devicemgmt_service()
+        now = datetime.now()
+        current = await device_mgmt.GetSystemDateAndTime()
+
+        dt_param = device_mgmt.create_type("SetSystemDateAndTime")
+        dt_param.DateTimeType = "Manual"
+        dt_param.DaylightSavings = current.DaylightSavings
+        dt_param.TimeZone = current.TimeZone
```

Comment:
Reuse the camera's existing time zone..

Reply:
That's something I'm trying to fix (it comes hardcoded to Mainland China)